pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -380,8 +380,8 @@ func TestPriorityQueue_MoveAllToActiveQueue(t *testing.T) {
 	addOrUpdateUnschedulablePod(q, q.newPodInfo(&unschedulablePod))
 	addOrUpdateUnschedulablePod(q, q.newPodInfo(&highPriorityPod))
 	q.MoveAllToActiveOrBackoffQueue("test")
-	if q.activeQ.Len() != 3 {
-		t.Error("Expected all items to be in activeQ.")
```

Comment:
Could you quickly explain why this changed?

Also, make it separate checks.
```
if a { t.Error}
if b { t.Error}
```

Reply:
previously the unschedule pods will be moved to activeQ if it doesn't exists in PodBackoffMap.
Now we should wait for backoff to complete or move the clock

Reply:
A few things:

- This test is titled: `MoveAllToActiveQueue`, which is probably just outdated.
- Why aren't pods being moved back to active? Don't they have 0 attempts? Do we care about 0?
- I suggest we grab the fake clock, set some timestamps and attempts in the pod infos, and make this test case more interesting. But... isn't this already covered in `TestPendingPodsMetric`?

Reply:
> Why aren't pods being moved back to active? Don't they have 0 attempts? Do we care about 0?

0 attempts means podInitialBackoffDuration

Originally, isPodBackingoff checked two things: pod exist in the PodBackoffMap and the backoff time is after ```clock.Now()```.  This unit test didn't properly add the pod to the unschedulable queue: it directly placed the pod in the queue instead of calling AddUnschedulableIfNotPresent, which means the pod didn't exist in PodBackoffMap, and so isPodBackingoff returned false and MoveAllToActiveOrBackoffQueue placed the pod in the active queue directly.

Now that we are basing isPodBackingoff only on backoff time, it return true, and so MoveAllToActiveOrBackoffQueue is placing the pods in the backoffQ.

Reply:
To avoid confusion, we should remove the `addOrUpdateUnschedulabePod` function. Also, it's worth noting that `AddUnschedulableIfNotPresent` doesn't increase the Attempts counter either. This is managed by the core scheduler itself.